### ISSUE TITLE
Second Midpoint Review - Accessibility Feedback - Accredited Representative Facing, Accredited Representative Portal, Form 21-22 approval denial

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#101091]
## Next Steps for the VFS team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://ows.io/qs/gb1dfnkl) at any time._

- [x] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [x] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [x] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [x] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Sorry for the delayed written feedback! Please feel free to reach out via Slack if you have any questions or concerns.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [x] **Must:** Spend some time thinking about focus management --- where keyboard focus goes after each action that a user takes. That's important for a number of reasons, but I'm especially thinking about screen reader users. When a screen reader user presses a button or performs some other action, you never want them to hear nothing at all. There should always be something that immediately flows from that action to indicate that (1) the action occurred and (2) what the new current state is.
   
   Thinking about the [search pattern](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Form-21-22-Design-Explorations?node-id=3575-24191&t=lz8pncoVPdxbKYd6-4), my expectation when searching by a postal code would be for focus to move to the "Showing X - Y of Z pending requests for [postal code]" message. That indicates that the search action occurred and describes the current state of the page. (It also matches focus behavior elsewhere on VA.gov, like in the sitewide search.) But that also means screen reader users searching by postal code will skip over everything between the search button and the status message. Some of that is okay to skip, eg. the link to search by individual instead. But some of it is directly relevant to the search they just performed, like whether they want pending or completed requests. And if you're not able to see that those elements exist, you would have no reason to go exploring the page to see what you may have jumped over.
   
   I don't have any specific recommendations for how it should be ordered (I don't know the product or the user base as well as you do!) but I would encourage you to map out the sequence in which a user is expected to perceive information and take actions, and to organize the page content to match that sequence.
- [x] **Coding note:** On a [POA request](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Form-21-22-Design-Explorations?node-id=3575-30827&t=lz8pncoVPdxbKYd6-4), my gut reaction is that the claimant information table and the authorization information table should be coded as lists styled to look like tables rather than actual HTML tables, since information isn't really complex enough to justify the semantic structure of a table. But if you do decide to go the actual-table route, note that the [va-table component](https://design.va.gov/storybook/?path=/docs/uswds-va-table--docs) expects column headers to be defined and a `table-title` prop, which is useful for screen reader users to decide if they want to engage with the table or not.
- [x] **Must:** On a [POA request](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Form-21-22-Design-Explorations?node-id=3575-30827&t=lz8pncoVPdxbKYd6-4), I'm a little worried the "We will send the claimant an email" info alert may be missed by screen reader users, eg. if a screen reader user presses [tab] after choosing their radio button and they skip straight to the submit button. I'd encourage you to look at using the `hint` prop on the `va-radio` component for this message, since the component is built to programmatically associate that text with the inputs (ensuring it's announced when tabbing into the group).
- [x] **Should:** Thinking about the structure of the [request cards](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Form-21-22-Design-Explorations?node-id=3671-108416&t=lz8pncoVPdxbKYd6-4), two recommendations:
   - Explore separating the link from the heading, eg. "Fitzgerald-Montgomery, Anderson" as a heading (with no link) and then a link in the card with text "View request for Fitzgerald-Montgomery, Anderson" or something similar. There's some evidence from usability testing in the MyHealtheVet application that screen reader users navigating by heading may miss that the headings are links and get confused about what their next steps should be. I don't think the evidence is solid enough for me to definitively say "don't do this" but I think it's worth being careful about.
   - Group the status label for each card in as a child of the heading, eg. `<h3><span class="some-class">Accepted</span> <span class="some-other-class">Fitzgerald-Montgomery, Anderson</span></h3>`. The children of the heading can be styled however you'd like, so visually the labels wouldn't look like heading text. But this structure helps make sure screen reader users navigating by heading don't miss the status label entirely.

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [ ] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone

**Comment by seipeljl** [comment]
> Must: Spend some time thinking about focus management --- where keyboard focus goes after each action that a user takes. That's important for a number of reasons, but I'm especially thinking about screen reader users. When a screen reader user presses a button or performs some other action, you never want them to hear nothing at all. There should always be something that immediately flows from that action to indicate that (1) the action occurred and (2) what the new current state is.
> Thinking about the [search pattern](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Form-21-22-Design-Explorations?node-id=3575-24191&t=lz8pncoVPdxbKYd6-4), my expectation when searching by a postal code would be for focus to move to the "Showing X - Y of Z pending requests for [postal code]" message. That indicates that the search action occurred and describes the current state of the page. (It also matches focus behavior elsewhere on VA.gov, like in the sitewide search.) But that also means screen reader users searching by postal code will skip over everything between the search button and the status message. Some of that is okay to skip, eg. the link to search by individual instead. But some of it is directly relevant to the search they just performed, like whether they want pending or completed requests. And if you're not able to see that those elements exist, you would have no reason to go exploring the page to see what you may have jumped over.  
> I don't have any specific recommendations for how it should be ordered (I don't know the product or the user base as well as you do!) but I would encourage you to map out the sequence in which a user is expected to perceive information and take actions, and to organize the page content to match that sequence.

**We will be exploring different search patterns in upcoming sprints, and will incorporate this feedback into those explorations.**


> Must: On a [POA request](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Form-21-22-Design-Explorations?node-id=3575-30827&t=lz8pncoVPdxbKYd6-4), I'm a little worried the "We will send the claimant an email" info alert may be missed by screen reader users, eg. if a screen reader user presses [tab] after choosing their radio button and they skip straight to the submit button. I'd encourage you to look at using the hint prop on the va-radio component for this message, since the component is built to programmatically associate that text with the inputs (ensuring it's announced when tabbing into the group).

**Done, we've eliminated the slim alert and moved this content to hint text.**

> Should: Thinking about the structure of the [request cards](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Form-21-22-Design-Explorations?node-id=3671-108416&t=lz8pncoVPdxbKYd6-4), two recommendations:
> 1) Explore separating the link from the heading, eg. "Fitzgerald-Montgomery, Anderson" as a heading (with no link) and then a link in the card with text "View request for Fitzgerald-Montgomery, Anderson" or something similar. There's some evidence from usability testing in the MyHealtheVet application that screen reader users navigating by heading may miss that the headings are links and get confused about what their next steps should be. I don't think the evidence is solid enough for me to definitively say "don't do this" but I think it's worth being careful about.

**This is a good callout, thank you. We'll be sure to keep an eye on this during pilot and adjust the card structure if needed.**

---

> 2) Group the status label for each card in as a child of the heading, eg. `<h3><span class="some-class">Accepted</span> <span class="some-other-class">Fitzgerald-Montgomery, Anderson</span></h3>`. The children of the heading can be styled however you'd like, so visually the labels wouldn't look like heading text. But this structure helps make sure screen reader users navigating by heading don't miss the status label entirely.

> Coding note: On a [POA request](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Form-21-22-Design-Explorations?node-id=3575-30827&t=lz8pncoVPdxbKYd6-4), my gut reaction is that the claimant information table and the authorization information table should be coded as lists styled to look like tables rather than actual HTML tables, since information isn't really complex enough to justify the semantic structure of a table. But if you do decide to go the actual-table route, note that the [va-table component](https://design.va.gov/storybook/?path=/docs/uswds-va-table--docs) expects column headers to be defined and a table-title prop, which is useful for screen reader users to decide if they want to engage with the table or not.

cc: @jquispe-oddball for visibility on these two items
